### PR TITLE
feat(components/progress-indicator): add help inline to progress indicator component

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2836540](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2836540)
[AB#2929277](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2929277)
[AB#2929282](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2929282)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2268/storybook/)

Component Storybooks:

- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/2268/storybooks/progress-indicator-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2268/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2268/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2268/playground/)

File: apps/code-examples/src/app/code-examples/progress-indicator/passive-indicator/basic/demo.component.html [hunk 1]
```diff
@@ -1,7 +1,13 @@
 <sky-progress-indicator [isPassive]="true" [startingIndex]="1">
-  <sky-progress-indicator-item title="Finished step"
+  <sky-progress-indicator-item
+    title="Finished step"
+    helpPopoverContent="Example help content for finished step"
+    helpPopoverTitle="Start here"
```
Comment (Blackbaud-SteveBrush) [inline apps/code-examples/src/app/code-examples/progress-indicator/passive-indicator/basic/demo.component.html:4]: Here, and in a few places, we need to ensure the inputs/outputs are alphabetized (especially in the code examples).

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts [hunk 2]
```diff
@@ -36,6 +37,31 @@ export class SkyProgressIndicatorItemComponent implements OnInit {
...
+   * A help key that identifies the global help content to display. When specified, a [help inline](https://developer.blackbaud.com/skyux/components/help-inline) button is
+   * placed beside the tree node label. Clicking the button invokes global help as configured by the application.
+   * @preview
+   */
+  @Input()
+  public helpKey: string | undefined;
+
+  /**
+   * The content of the help popover. When specified along with `labelText`, a [help inline](https://developer.blackbaud.com/skyux/components/help-inline)
+   * button is added to the checkbox group fieldset legend. The help inline button displays a [popover](https://developer.blackbaud.com/skyux/components/popover)
```
Comment (Blackbaud-SteveBrush) [inline libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: "checkbox group"

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts [hunk 3]
```diff
@@ -36,6 +37,31 @@ export class SkyProgressIndicatorItemComponent implements OnInit {
     return this.#_title;
   }
 
+  /**
+   * A help key that identifies the global help content to display. When specified, a [help inline](https://developer.blackbaud.com/skyux/components/help-inline) button is
+   * placed beside the tree node label. Clicking the button invokes global help as configured by the application.
```
Comment (Blackbaud-SteveBrush) [inline libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: "tree node"

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.html [hunk 4]
```diff
@@ -12,21 +12,35 @@
...
+      </span>
+      @if (helpKey || helpPopoverContent) {
+        <sky-help-inline
+          [helpKey]="helpKey"
+          [labelText]="title"
+          [popoverTitle]="helpPopoverTitle"
+          [popoverContent]="helpPopoverContent"
+        />
+      } @else {
+        <span class="sky-control-help-container" skyTrim
```
Comment (Blackbaud-SteveBrush) [inline libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.html]: Consumer unit tests may be expecting this element to be a child of `.sky-progress-indicator-item-heading`. I'd put this back where it was, but wrap it in reverse logic:
```
@if (!helpKey && !helpPopoverContent) {
```

File: apps/code-examples/src/app/code-examples/progress-indicator/passive-indicator/basic/demo.component.html (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply apps/code-examples/src/app/code-examples/progress-indicator/passive-indicator/basic/demo.component.html:4]: reordered

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.html (hunk 4 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.html]: moved

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts (hunk 3 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: fixed

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: fixed

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**